domaincommon/src/state.rs

```diff
@@ -1023,8 +1124,130 @@ mod test {
+            mech: "al".to_string(),
+        });
+        cmds.push(MechCommand::Move {
+            turn: 0,
+            direction: GridDirection::NorthEast,
+            mech: "bob".to_string()
+        });
+        cmds.push(MechCommand::Move {
+            turn: 0,
+            direction: GridDirection::NorthEast,
+            mech: "bob".to_string()
+        });
+        cmds.push(MechCommand::FirePrimary {
+            turn: 0,
+            direction: GridDirection::NorthEast,
+            mech: "bob".to_string(),
+        });
+        cmds.push(MechCommand::Move {
+            turn: 0,
+            direction: GridDirection::SouthWest,
+            mech: "bob".to_string()
+        });
+        cmds.push(MechCommand::FinishTurn {
+            turn: 0,
+            mech: "al".to_string(),
+        });
+        cmds.push(MechCommand::FinishTurn {
+            mech: "bob".to_string(),
+            turn: 0,
+        });
+
+        let state = cmds.iter().fold(state, |state, cmd| {
+            Match::handle_command(&state, &cmd)
+                .unwrap()
+                .iter()
+                .fold(state, |state, evt| Match::apply_event(&state, evt).unwrap())
+        });
+
+        assert_eq!(
+            state.mechs["bob"].health,
+            INITIAL_HEALTH - SECONDARY_SPLASH_DAMAGE
+        );
+
+        let al_position = Point::new(10, 6);
+        let bob_position = Point::new(18, 13);
+        assert_eq!(
+            state.mechs["al"].position.x,
+            al_position.x
+        );
+        assert_eq!(
+            state.mechs["al"].position.y,
+            al_position.y
+        );
+        assert_eq!(
+            state.mechs["bob"].position.x,
+            bob_position.x
+        );
+        assert_eq!(
+            state.mechs["bob"].position.y,
+            bob_position.y
+        );
+    }
+
+    #[test]
+    fn action_points_limit_prevents_death() {
+        let state = gen_root_state(
+            vec![("al", Point::new(10, 6)), ("bob", Point::new(11, 6))], // Bob should be within splash range
+            2,
+        );
+
+        let mut cmds = Vec::new();
+        for i in 0..20 {
+            cmds.push(MechCommand::FirePrimary {
+                turn: i,
+                direction: GridDirection::East,
+                mech: "al".to_string(),
+            });
+        }
+        cmds.push(MechCommand::FinishTurn {
+            turn: 0,
+            mech: "al".to_string(),
+        });
+        cmds.push(MechCommand::FinishTurn {
+            mech: "bob".to_string(),
+            turn: 0,
+        });
+
+        let state = cmds.iter().fold(state, |state, cmd| {
+            Match::handle_command(&state, &cmd)
+                .unwrap()
+                .iter()
+                .fold(state, |state, evt| Match::apply_event(&state, evt).unwrap())
+        });
+
+        assert_eq!(
+            state.mechs["bob"].health,
+            INITIAL_HEALTH - ( PRIMARY_DAMAGE * 2 )
+        );
+
+
+    }
```

Comment:
Al and Bob truly should find a better way to work out their differences than firing projectiles at each other.